### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `scroll-state` on `container-type`

### ISSUE THREAD

**Issue by n0099** [stylelint/stylelint#8607]
### What minimal example or steps are needed to reproduce the bug?

```vue
<style scoped>
.my-class {
    container-type: scroll-state;
}
</style>
```

### What minimal configuration is needed to reproduce the bug?

```ts
import type { Config } from 'stylelint';

export default {
    extends: [
        'stylelint-config-recommended',
        'stylelint-config-recommended-vue'
    ]
} as Config;
```

### How did you run Stylelint?

```
stylelint '**/*.vue'
```

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "^16.20.0",
    "stylelint-actions-formatters": "^16.3.1",
    "stylelint-config-recommended-vue": "^1.6.0"
}
```

### What did you expect to happen?

Not reporting as https://drafts.csswg.org/css-conditional-5/#valdef-container-type-scroll-state

### What actually happened?

```
test.vue
  1:21  ✖  Unexpected unknown value "scroll-state" for property "container-type"  declaration-property-value-no-unknown
```

**Comment by jeddy3** [comment]
@n0099 Thanks for the report and for using the template.

This will be fixed when the syntax is updated in our upstream parser.

In the meantime, you can extend the syntax yourself using the [`languageOptions`](https://stylelint.io/user-guide/configure#languageoptions) configuration property:

```json
{
  "languageOptions": {
    "syntax": {
      "properties": { 
        "container-type": "| scroll-state"
      }
    }
  },
  "extends": [
    "stylelint-config-standard"
  ]
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdiymAuSAlpjAE4C0eAngA4zwoDOYpEANm+YwXjANxYAviAA0IAGaE2MAHJIAtnEQwAHgprSAdGEaNR4CJkkBzBCAzYU6EGySZjAVyTGYAeRp5ChxtYYWUAVYgjFT4SCq+aFiBgdY0rHSknjA+CGhWljEB1rgExGSUtDCR1gA+TCzsnNxIvNbRWYINAU2WgiIN1qq8mFCpDADazUHcVNJsxHjkuCZcBL1IpFD1lgC6QvozhMYAYhCk8rVmAFaMhvqwNHqI-iPU45MlNrUpeNYdltajD-jThrM1BZLJ62XjcFbCQRAA)
